### PR TITLE
Add snapcraft.yaml docs

### LINKED ISSUE #11: Add proper documentation
The `snapcraft.yaml` should be modified to have the metadata documentation rather than relying on the documentation provided through the store.

### PR DESCRIPTION
**Pull request by Deezzir**
Closes: #11

### CODE REVIEW

File: snap/snapcraft.yaml
Comment (samuelallan72) [inline snap/snapcraft.yaml:11]: Snap store markdown doesn't support headings?

File: snap/snapcraft.yaml
Comment (samuelallan72) [inline snap/snapcraft.yaml:24]: Are backslashes necessary? Does snap store markdown auto reflow paragraphs?

File: snap/snapcraft.yaml
Comment (samuelallan72) [inline snap/snapcraft.yaml]:
```suggestion
  By default the openstack-exporter services serves on port localhost:9180 at the /metrics URL:
```

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:11]: Nope, the supported markdown is really lean
image

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:24]: This way I can force a new line, otherwise it will be one line

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:11]: You can check the dcgm's docs and the store page for reference.
https://snapcraft.io/dcgm
https://github.com/canonical/dcgm-snap/blob/main/snap/snapcraft.yaml

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:24]: Looks better on the store :)

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:11]: It's annoying

Comment (samuelallan72) [reply snap/snapcraft.yaml:11]: Ah yep sorry, I did just check that. :+1:

File: snap/snapcraft.yaml
Comment (samuelallan72) [reply snap/snapcraft.yaml:24]: Oh that forces a newline, ok. :+1: Could you use a new paragraph instead? Or does that mess up the flow?

Comment (Deezzir) [reply snap/snapcraft.yaml:24]: I prefer that way. Otherwise, each sentence will be separated by two new lines (in the code). Example:

```
 The openstack-exporter service is disabled by default and needs to be configured before the startup.\
 For example:
 ```
 
 ```
  The openstack-exporter service is disabled by default and needs to be configured before the startup.
  
  For example:
  ```
  
Both will give the same results, but with `\`, I can group them, just a preference

Comment (samuelallan72) [reply snap/snapcraft.yaml:24]: Ah I see. It feels more manageable in the code with a new paragraph rather than the backslash. It's confusing that backslash forces a linebreak; usually backslash at the end of the line *escapes* the linebreak so you get one line. Yaml [multiline strings are complicated](https://yaml-multiline.info/index.html). :/

(this is just a nit though)

**Review by samuelallan72 — APPROVED** [review]

File: snap/snapcraft.yaml
Comment (jneo8) [inline snap/snapcraft.yaml:49]: **question (non-blocking):** 

Does it feel necessary to teach users how to configure a boolean value in this section? To me, it seems more fitting for the official documentation on *snapcraft.io*.

**Review by jneo8 — APPROVED** [review]
Approved, as everything looks good overall. LGTM!

I have one non-blocking question. Feel free to resolve.

**Comment by Deezzir** [comment]
@aieri will wait for your final review!

File: snap/snapcraft.yaml
Comment (aieri) [inline snap/snapcraft.yaml:18]: nit: should this be a `must`? Or are you perhaps trying to convey that not everyone will need to read from `/etc/openstack`?

File: snap/snapcraft.yaml
Comment (aieri) [reply snap/snapcraft.yaml:49]: yes, I agree

**Review by aieri — APPROVED** [review]
+1, with suggestions

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:18]: I was referencing the upstream docs for that, maybe there is a point being made that there can be an autoconnect

File: snap/snapcraft.yaml
Comment (Deezzir) [reply snap/snapcraft.yaml:49]: I think it depends on the implementation. It does not have to be `true` or `false`. It can be something else like yes/no or 0/1. We choose to use these values which are parsed not by snapd but by our configure script

**Pull request merged**
